Experiments_toolkit.py

```diff
@@ -57,151 +50,143 @@ def splitting_up_data(dictionary, dict_key, train, percent_train, test, percent_
+    neg_label = np.zeros(len(neg))
+    labels = np.concatenate((pos_label, neg_label))
+    dataset = tfdata.Dataset.from_tensor_slices((examples, labels))
+    return dataset
+
+
+def preprocess_add_augmentation(data, augmentation_function, sigma):
     """
        Description: Iterate over DATA and proprocess the traces by adding an augmentation function
                                 to the data.
 
        Args:
-           DATA -- np array (Numpy array where the last row is time steps)
+           data -- list (List of numpy array where the last row is time steps, [(sensor_dims + 1), num_samples])
            dictionary -- dict (Dictionary to store augmented traces)
            pos_key -- string (Key in dictionary to store augmented windows) 
            augmentation_function -- function (Desired augmentation function of the form:
-                                                                                        augment(np_array data, int sigma))
+                                              augment(np_array data, int sigma))
            sigma -- double (Level of augment to be passed to the augmentation function )
            size -- int (Window size of traces to be stored in dictionary)
 
        Return:
            dictionary -- dict (Dictionary with augmented windows of size = size in key pos_key)
      """
-    for j in range(len(DATA)):
-        sample = DATA[j]
-        length = len(np_sample[0][0])
-        if length < size:
-            continue
-        window = same_window_size(sample[:-1], size, 10)
-        transpose = window.transpose(0, 2, 1)
-        for i in range(len(transpose)):
-            augmented = augmentation_function(transpose[i], sigma)
-            dictionary[pos_key].extend([augmented])
-
-    return dictionary
-
-
-def DA_Rotation_specific(X, angle_low, angle_high, axis):
-    axis = np.array(axis)
-    angle = np.random.uniform(low=angle_low, high=angle_high)
-    return np.matmul(X, axangle2mat(axis, angle))
-
-
-def scatter_PCA(X, Y, alpha):
-    pca = PCA(n_components=3)
-    pca_result = pca.fit_transform(X)
-    scatter_plot(pca_result, Y, alpha)
-
-
-def scatter_ICA(X, Y, alpha):
-    ica = FastICA(n_components=3)
-    ica_result = ica.fit_transform(X)
-    scatter_plot(ica_result, Y, alpha)
-
-
-def scatter_TSNE(X, Y, alpha):
-    RS = 20150101
-    TSNE_proj = TSNE(random_state=RS, n_components=3).fit_transform(X)
-    scatter_plot(TSNE_proj, Y, alpha)
-
-
-def scatter_plot(result, Y, alpha):
-    df_subset = {"negative": [], "positive": []}
-    df_subset['positive'] = result[:, 0]
-    df_subset['negative'] = result[:, 1]
-    df_subset['y'] = Y
-    plt.figure(figsize=(10, 10))
-    sns.scatterplot(
-        x="negative", y="positive",
-        hue="y",
-        palette=sns.color_palette("hls", 2),
-        data=df_subset,
-        legend="full",
-        alpha=alpha
-    )
+    augmented = []
+    for j in range(len(data)):
+        sample = data[j]
+        sample = sample[:-1]
```

Comment:
does that remove the time dimension? Can you rename the variable to explain what happens?

Reply:
Yep!